offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/product/MaintainProductsViewModel.groovy

```diff
@@ -46,4 +50,12 @@ class MaintainProductsViewModel {
         products.clear()
         products.addAll(productsResourcesService.iterator().toList())
     }
+
+    Product getSelectedProduct() {
+        if(selectedProduct.isPresent()) {
+            return selectedProduct.get()
+        } else {
+            throw new RuntimeException("No product is currently selected.")
+        }
+    }
```

Comment:
I think you don't need it. Because you have an optional you will get a nullpointer when trying to get it. I also think it's not good to have unhandled exceptions.

```suggestion

```

Reply:
Good Idea, however the main point here is to ensure that we don't get nullpointerexceptions in the code down the line, since we have to get the product from the optional container at multiple instances.

Reply:
But since it is an Optional. You can handle the nullpointer at the point it occurs while you try to get the product.

Reply:
Alrighty